source/developer-guides/debug.rst

```diff
@@ -0,0 +1,112 @@
+.. _debug:
+
+Source Level Debugging on Serial Port
+--------------
+
+This explains how to perform basic debugging operation with Intel UDK Debugger on SBL over serial port.
+
+**Prerequisite**
+
+Install Intel UDK Debugger on HOST::
+Intel UDK Debugger can be downloaded here for Windows and Linux, and just install it with default configuration values.
+
+https://firmware.intel.com/develop/intel-uefi-tools-and-utilities/intel-uefi-development-kit-debugger-tool#overlay-context=develop
+
+https://firmware.intel.com/sites/default/files/UDK_Debugger_Tool_User_Manual_V1.11.pdf
+
+
+**General steps**
+
+1. Enable DebugAgent on SBL and flash the SBL image(or IFWI) on TARGET
+
+2. Configure Intel UDK Debugger environment on HOST
+
+3. Launch Intel UDK Debugger on HOST
+
+4. Turn TARGET on
+
+5. Start debugging
+
+
+**Step 1 - Enable Debug Agent**
+
+By default, DebugAgent is not enabled. It can be simply enabled with a single line change.
+
+1. Open BoardConfig.py of specific platform, set **ENABLE_SOURCE_DEBUG** to **1**::
+
+    self.ENABLE_SOURCE_DEBUG = 1
+  
+2. APL platform has a known issue on Stage1A debugging because of its size limitation. ONLY APL platform requires **SKIP_STAGE1A_SOURCE_DEBUG** to **1**::
```

Comment:
I think we should not say "ONLY" APL here. It may be the case that future platforms have such a restriction as well.